kickstarter/android-oss
# NTV-270: Extend image parsing to handle captions, links

# 📲 What

- The parser now is able to detect Images with captions and links

# 🛠 How
When the parser detects an image/gif, or an image/gif with caption, or an image/gif with a link, or an image/gif with caption and Link it will return a `ImageViewElement` with the populated fields according to each case: 
`data class ImageViewElement(
    val src: String,
    val href: String? = null,
    val caption: String? = null
) : ViewElement`

# 👀 See

<img width="1264" alt="ImageParser" src="https://user-images.githubusercontent.com/4083656/141027648-3c5b130d-86d8-4721-bb36-8438dbe412a5.png">
| --- | --- |
|  |  |

# 📋 QA
No user facing changes
- make sure the list of elements the parser returns does not contain any duplicate.
- make sure if the project html contains an image or gif with or without caption and with or without link the ImageViewElement on each case has the correct fields populated.

# Story 📖

[NTV-270](https://kickstarter.atlassian.net/browse/NTV-270)
